mono/metadata/domain.c

```diff
@@ -590,20 +590,27 @@ mono_init_internal (const char *filename, const char *exe_filename, const char *
 		mono_fixup_exe_image (exe_image);
 #endif
 	} else if (runtime_version != NULL) {
-		runtimes = g_slist_prepend (runtimes, (gpointer)get_runtime_by_version (runtime_version));
+		const MonoRuntimeInfo* rt = get_runtime_by_version (runtime_version);
+		if (rt != NULL)
```

Comment:
Should we not warn here as well?

Reply:
If `rt` is null here, `runtimes` will be null, so we will fall down into the default runtime code which will warn.

That is, `mono_init_version ("pedump", "v2.0.50727")` will print out `WARNING: The requested runtime version "v2.0.50727" is unavailable`.

Reply:
Ah okay, misread surrounding code.